src/main/java/store/onlinebookstore/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,67 @@
+package store.onlinebookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import store.onlinebookstore.dto.cartitem.CreateCartItemRequestDto;
+import store.onlinebookstore.dto.cartitem.ItemQuantity;
+import store.onlinebookstore.dto.shoppingcart.ShoppingCartDto;
+import store.onlinebookstore.service.ShoppingCartService;
+
+@Tag(name = "ShoppingCart management", description = "Endpoints for managing shopping cart")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/cart")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping
+    @Operation(summary = "Show shopping cart",
+            description = "Show shopping cart for logged in user")
+    public ShoppingCartDto getShoppingCartForUser(Authentication authentication) {
+        String email = authentication.getName();
+        return shoppingCartService.findByEmail(email);
```

Comment:
For all methods.
Id is a primary key in users table. Search by  primary key is the best way - we always have index on DB side.
```suggestion
        User user = (User) authentication.getPrincipal();
        return shoppingCartService.getById(user.getId());
```